### ISSUE TITLE
mypy does not type check arguments to `Annotated`

### ISSUE THREAD

**Issue by eltoder** [python/mypy#19438]
**Bug Report**

Mypy does not type check arguments to the `typing.Annotated` special form. While mypy is not expected to understand the _meaning_ of these arguments or have them affect the resulting type, it should still check that the expressions are well formed, since they are normal executable Python code, the same as everything else.

**To Reproduce**

```python
from dataclasses import dataclass
from typing import Annotated

class Module:
    @dataclass
    class Metadata:
        value: str

X1 = Annotated[int, Module.Metadata(v="val")]  # wrong keyword
X2 = Annotated[int, Module.Metadata(123)]  # wrong arg type
X3 = Annotated[int, Module.Typodata("123")]  # wrong class
```
https://mypy-play.net/?mypy=1.16.1&python=3.12&gist=ada2444e90499fc01c745acdb53ed771

**Expected Behavior**

The incorrect uses of `Metadata` in `X1`, `X2`, and `X3` should generate errors. They do if I remove `Annotated`:
```
main.py:9: error: Unexpected keyword argument "v" for "Metadata"  [call-arg]
main.py:10: error: Argument 1 to "Metadata" has incompatible type "int"; expected "str"  [arg-type]
main.py:11: error: "type[Module]" has no attribute "Typodata"  [attr-defined]
```

**Actual Behavior**

No errors.

**Your Environment**

- Mypy version used: 1.16.1
- Python version used: 3.12

**Comment by brianschubert** [comment]
Thanks, closing as duplicate of https://github.com/python/mypy/issues/10872
